codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -120,9 +118,54 @@ If there are obsolete binaries in entitlements configuration files, please delet
 update these file paths accordingly.
 ''';
 
+  List<String> engineArtifactFilePaths = [
+    "android-arm-profile/darwin-x64.zip",
+    "android-arm-release/darwin-x64.zip",
+    "android-arm64-release/darwin-x64.zip",
+    "android-arm64-profile/darwin-x64.zip",
+    "android-x64-profile/darwin-x64.zip",
+    "android-x64-release/darwin-x64.zip",
+    "dart-sdk-darwin-arm64.zip",
+    "dart-sdk-darwin-x64.zip",
+    "darwin-x64/artifacts.zip",
+    "darwin-x64-profile/artifacts.zip",
+    "darwin-x64-release/artifacts.zip",
+    "darwin-x64/gen_snapshot.zip",
+    "darwin-x64-profile/gen_snapshot.zip",
+    "darwin-x64-release/gen_snapshot.zip",
+    "darwin-x64/font-subset.zip",
+    "darwin-x64/FlutterMacOS.framework.zip",
+    "darwin-x64-profile/FlutterMacOS.framework.zip",
+    "darwin-x64-release/FlutterMacOS.framework.zip",
+    "ios/artifacts.zip",
+    "ios-profile/artifacts.zip",
+    "ios-release/artifacts.zip",
+  ];
+
   /// The entrance point of examining and code signing an engine artifact.
   Future<void> validateAll() async {
-    await Future<void>.value(null);
+    List<String> artifactFilePaths = <String>[];
+    if (googleCloudStorage.optionalSwitch.isNotEmpty) {
+      artifactFilePaths = googleCloudStorage.optionalSwitch;
+    } else {
+      artifactFilePaths = engineArtifactFilePaths;
+    }
+
+    final Iterable<Future<void>> futures = artifactFilePaths.map((String artifactFilePath) {
+      final Directory outDir =
+          rootDirectory.childDirectory('remote_zip_${artifactFilePath.replaceAll("/", "_").replaceAll(".zip", "")}');
+      return processRemoteZip(artifactFilePath: artifactFilePath, parentDirectory: outDir);
+    });
+    await Future.wait(
+      futures,
+      eagerError: true,
+    );
+
+    if (dryrun) {
+      log.info('code signing dry run has completed, If you intend to upload the artifacts back to'
+          ' google cloud storage, please use the --dryrun=false flag to run code signing script.'
+          ' thanks for understanding the security concerns!');
```

Comment:
I don't think we need this line 🙂

Reply:
umm I added it here because of https://github.com/flutter/flutter/issues/104387 and https://github.com/flutter/flutter/issues/104450#event-6664083158. (caused by me uploading corrupted artifacts back to gcloud, and created havoc for people). So by design codesign app won't upload artifacts back to gcloud. Since users might want to upload artifacts back instead (to complete the cycle of code signing), I added a prompt saying users could use --dryrun=false to state that they are ready to upload the artifacts back to gcloud (and inform users they would be responsible for it)

Reply:
The "thanks for understanding the security concerns!" is not needed

Reply:
Oh I see, sure, removed!